src/Microsoft.AspNet.Cors.Core/CorsServiceCollectionExtensions.cs

```diff
@@ -0,0 +1,45 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.AspNet.Cors;
+using Microsoft.AspNet.Cors.Core;
+using Microsoft.Framework.ConfigurationModel;
+using Microsoft.Framework.Internal;
+
+namespace Microsoft.Framework.DependencyInjection
+{
+    /// <summary>
+    /// The <see cref="IServiceCollection"/> extensions for enabling CORS support.
+    /// </summary>
+    public static class CorsServiceCollectionExtensions
+    {
+        /// <summary>
+        /// Can be used to configure services in the <paramref name="serviceCollection"/>.
+        /// </summary>
+        /// <param name="serviceCollection">The service collection which needs to be configured.</param>
+        /// <param name="configure">A delegate which is run to configure the services.</param>
+        /// <returns></returns>
+        public static IServiceCollection ConfigureCors(
+            [NotNull] this IServiceCollection serviceCollection,
+            [NotNull] Action<CorsOptions> configure)
+        {
+            return serviceCollection.Configure(configure);
+        }
+
+        /// <summary>
+        /// Add services needed to support CORS to the given <paramref name="serviceCollection"/>.
+        /// </summary>
+        /// <param name="serviceCollection">The service collection to which CORS services are added.</param>
+        /// <param name="config">An optional configuration which can be applied to the <see cref="ServiceDescriber"/>.
+        /// </param>
+        /// <returns>The updated <see cref="IServiceCollection"/>.</returns>
+        public static IServiceCollection AddCors(this IServiceCollection serviceCollection, IConfiguration config = null)
+        {
+            var describe = new ServiceDescriber(config);
```

Comment:
@davidfowl fyi `ServiceDescriber` hasn't yet left the building.  I should be pushing the MVC PR (#2137) in a bit.

@harshgMSFT this code should just do `serviceCollection.AddTransient<ICorsService, CorsService>();` or perhaps `TryAdd(anotherCollection)` where `var anotherCollection = new ServiceCollection();`.

in addition, you shouldn't need an overload accepting an `IConfiguration` parameter.